sky/spot/recovery_strategy.py

```diff
@@ -218,7 +221,7 @@ def _launch(self, max_retry=3, raise_on_failure=True) -> Optional[float]:
                     continue
 
                 # Check the job status until it is not in initialized status
-                if status is not None and job_lib.JobStatus.PENDING < status:
+                if status is not None and job_lib.JobStatus.INIT < status:
```

Comment:
This seems a bit weird. 
- JobStatus = [INIT, SETTING_UP, PENDING, RUNNING, ...]
- SpotStatus = [PENDING, SUBMITTED, STARTING, RUNNING, ...]

With this new change, this means we could return to controller.py from launch() when the JobStatus is in SETTING_UP or PENDING.  The controller process will [then set](https://github.com/skypilot-org/skypilot/blob/c6cd1d64df2e1afb6d9c3922885adae970170e1a/sky/spot/controller.py#L69) the SpotStatus to RUNNING.

Reply:
Yes, I think that might be fine, as we can modify the definition of spot job RUNNING to actual job `SETTING_UP` or `RUNNING`. That means the `spot_status` only shows the status of whether user's commands start running. The `PENDING` status for the actual job will not last for a long time, since the spot cluster launched is dedicated to the job, i.e. the job will be RUNNING as soon as the `SETTING_UP` finish.

Reply:
We may need to have a very detailed comment, incl the two lists in my previous comment + some tricky behaviors, somewhere. Maybe spot_state?

Reply:
Good point! I added the comments in the `spot_state.SpotStatus`. PTAL : )